### PR TITLE
Copy metadata for line number aware sources

### PR DESCRIPTION
**Pull request by bill-rich**
Multiple results from the same chunk could result in line numbers in the metadata being overwritten. Deep copy the chunk and source metadata to avoid clobbering.

### CODE REVIEW

File: pkg/engine/engine.go [hunk 1]
```diff
@@ -230,9 +230,20 @@ func (e *Engine) detectorWorker(ctx context.Context) {
...
 						for _, result := range results {
-							SetResultLineNumber(chunk, &result, fragStart, mdLine)
+							resultChunk := chunk
+							if isGitSource(chunk.SourceType) {
+								copyChunk := *chunk
+								copyMetaDataClone := proto.Clone(chunk.SourceMetadata)
+								copyMetaData, ok := copyMetaDataClone.(*source_metadatapb.MetaData)
+								if ok {
+									copyChunk.SourceMetadata = copyMetaData
+								}
```
Comment (mcastorina) [inline pkg/engine/engine.go:240]: **Nit (optional):** These lines can be combined which provides limited scope to the variables

```go
if copyMetaData, ok := copyMetaDataClone.(*source_metadatapb.MetaData); ok {
    copyChunk.SourceMetadata = copyMetaData
}
```

File: pkg/engine/engine.go [hunk 2]
```diff
@@ -314,8 +325,6 @@ func FragmentFirstLine(chunk *sources.Chunk) (int64, *int64) {
 
 // SetResultLineNumber sets the line number in the provided result.
 func SetResultLineNumber(chunk *sources.Chunk, result *detectors.Result, fragStart int64, mdLine *int64) {
-	if isGitSource(chunk.SourceType) {
-		offset := FragmentLineOffset(chunk, result)
-		*mdLine = fragStart + offset
-	}
+	offset := FragmentLineOffset(chunk, result)
+	*mdLine = fragStart + offset
 }
```
Comment (mcastorina) [inline pkg/engine/engine.go:330]: I think `SetResultLineNumber` gets called in other places, so we may need to keep this `isGitSource` check.

**Review by mcastorina — COMMENTED** [review]
Overall looks good, just one concern about `SetResultLineNumber`.

File: pkg/engine/engine.go (hunk 2 above)
Comment (bill-rich) [reply pkg/engine/engine.go:330]: It does also in thog, but that'll need the same changes to fix line numbers as here.

**Review by mcastorina — APPROVED** [review]

**Pull request merged**